### PR TITLE
pkg/qDSA: fix backend selection log

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

qDSA as optimized implementations for AVR. The logic to select the optimized
backend predates the architecture features and uses some hand crafted logic.
This updates the logic to select the backend based on the boards architecture
using the feature system instead.

### Testing procedure

No changes in generated binaries.

### Issues/PRs references

Fixes compilation with ATXmega boards introduced in https://github.com/RIOT-OS/RIOT/pull/15758

### CODE REVIEW

**Comment by benpicco** [comment]
The test `tests/pkg_qdsa` does now crash on `same54-xpro`:

```
2021-03-08 15:30:29,577 # main(): This is RIOT! (Version: 2021.04-devel-979-g22406-HEAD)
2021-03-08 15:30:29,578 # .
2021-03-08 15:30:29,580 # Context before hardfault:
2021-03-08 15:30:29,581 #    r0: 0x20001250
2021-03-08 15:30:29,583 #    r1: 0x1bebddb6
2021-03-08 15:30:29,585 #    r2: 0x20001240
2021-03-08 15:30:29,586 #    r3: 0x200011e0
2021-03-08 15:30:29,588 #   r12: 0x20001130
2021-03-08 15:30:29,589 #    lr: 0x000043bd
2021-03-08 15:30:29,591 #    pc: 0x000014c4
2021-03-08 15:30:29,592 #   psr: 0x01000200
2021-03-08 15:30:29,593 # 
2021-03-08 15:30:29,593 # FSR/FAR:
2021-03-08 15:30:29,595 #  CFSR: 0x01000000
2021-03-08 15:30:29,597 #  HFSR: 0x40000000
2021-03-08 15:30:29,598 #  DFSR: 0x00000000
2021-03-08 15:30:29,600 #  AFSR: 0x00000000
2021-03-08 15:30:29,600 # Misc
2021-03-08 15:30:29,602 # EXC_RET: 0xfffffffd
2021-03-08 15:30:29,604 # Active thread: 1 "main"
2021-03-08 15:30:29,608 # Attempting to reconstruct state for debugging...
2021-03-08 15:30:29,609 # In GDB:
2021-03-08 15:30:29,610 #   set $pc=0x14c4
2021-03-08 15:30:29,611 #   frame 0
2021-03-08 15:30:29,612 #   bt
2021-03-08 15:30:29,612 # 
2021-03-08 15:30:29,615 # ISR stack overflowed by at least 16 bytes.
```

File: pkg/qDSA/Makefile.dep [hunk 1]
```diff
@@ -1,8 +1,8 @@
-ifneq (,$(filter cortex-m23 cortex-m0%,$(CPU_CORE)))
+ifneq (,$(filter arch_arm_cortex_m,$(FEATURES_USED)))
```
Comment (aabadie) [inline pkg/qDSA/Makefile.dep:1]: Why not re-use the `cpu_core_cortexm` feature ? It seems to have the same purpose.
`

Comment (maribu) [reply pkg/qDSA/Makefile.dep:1]: Why do some arch features are prefixed `arch_` and others are `cpu_core_`? Maybe we should do an `s/cpu_core_cortexm/arch_arm_cortex_m/g`?

**Comment by maribu** [comment]
> The test `tests/pkg_qdsa` does now crash on `same54-xpro` (Cortex M4F):

Yes, indeed. Previously the code was only used for a subset of Cortex-M CPUs. Let me revert this and only use the feature for AVR.

(Btw.: The optimized version requires significantly more ROM. Could it be, that the optimized version predates proper optimizing C compilers, and that the C version now is preferable?)

File: pkg/qDSA/Makefile.dep (hunk 1 above)
Comment (aabadie) [reply pkg/qDSA/Makefile.dep:1]: Indeed but homogenizing this is another PR. In the context of this PR, there's no need to introduce another feature if there's already one that does the same job.

Comment (maribu) [reply pkg/qDSA/Makefile.dep]: In addition, I overlooked that the selection indeed depends on the specific Cortex M is build for. So the approach taken here doesn't work here anyway for the ARM selection.

**Comment by maribu** [comment]
With the Cortex-M stuff being dropped, this is now minor again :-D

**Comment by benpicco** [comment]
I mean you can usually trade ROM size for speed with look-up-tables and loop unrolling (unless there is a significant cache advantage).
I'm just a bit puzzled why the Cortex-M0 code fails on Cortex-M4, I thought M4 is a superset of M0.

**Review by benpicco — APPROVED** [review]

**Comment by benpicco** [comment]
Tested on `avr-rss2`

**Pull request merged**